conda_lock/conda_lock.py

```diff
@@ -165,21 +165,25 @@ def print_proc(proc):
     try:
         proc.check_returncode()
     except subprocess.CalledProcessError:
-        try:
-            err_json = json.loads(proc.stdout)
-            message = err_json["message"]
-        except json.JSONDecodeError as e:
-            print(f"Failed to parse json, {e}")
-            message = ""
-
         print(f"Could not perform conda install using {file} lock file into {prefix}")
-        if message:
-            print(message)
+        print(_handle_subprocess_stdout(proc.stdout))
         print_proc(proc)
-
         sys.exit(1)
 
 
+def _handle_subprocess_stdout(stdout, retry=True):
+    try:
+        err_json = json.loads(stdout)
+        return "\n\n".join(error["message"] for error in err_json["errors"])
+    except json.JSONDecodeError as e:
+        if retry:
+            return _handle_subprocess_stdout(
+                "\n".join(stdout.split("\n")[1:])[1:], retry=False
```

Comment:
`JSONDecodeError`s seem to happen for `conda` but not for `mamba`.  In `tests/test-stdout/conda.txt` you can see that the `proc.stdout` for a run with `conda` return a separate JSON object in the first line (that one we want to ommit) and a strangely decoded character in the first position of the second line (that we want to ommit as well).

After stripping the first line and the first character of the second line, you can parse the `proc.stdout` to JSON.

Unfortunately I haven't seen enough error messages to see if this applies generally. If you have more examples for errors, I'd love to add them to the tests. 

Another thing to try would be to provide `proc.stderr` instead of `proc.stdout`, but I haven't checked yet how that looks like.

Reply:
Okay you can actually do this much easier. So the `proc.stdout` contains null terminated strings (`\x00`). You can split by that character and take the last element of the list. This also handles cases where there are multiple single line JSON object on top of the error object (which can happen, I have updated the `conda.txt` test file accordingly).

This way we also get rid of the pesky recursion that I put there.